assessments/projects/openfga/joint-assessment.md

```diff
@@ -0,0 +1,647 @@
+|MITRE classification|TA0001: Initial Access -> T1195: Supply Chain Compromise|
+|Actors|openfga.server|
+|Suggested Mitigation|Pin the dependency using SHA tag for the container image.|
+|Impact (High/ Med/ Low)|Low|
+|Likelihood (High/ Med/ Low)|Low|
+
+### Other findings
+
+This section has other findings that could not be classified 
+in earlier parts. It includes exploits such as server DDOS and 
+potential leakage of information about application landscape.
+
+|Summary|Playground link as well as shape of API identifies the openfga server.|
+|--|--|
+|Discovered in self-assessment?|No|
+|Weakness|An attacker performing reconnaissance will be able to identify if a service is the openfga server, a high-value target.|
+|MITRE classification|TA0043: Reconnaissance -> T1592: Gather Victim Host Information|
+|Actors|openfga.server|
+|Suggested Mitigation|Not sure if there is a remediation, given it’s the case for any API server that endpoints will return a 403 rather than 404 for unauthenticated access. An attacker iterating through the expected shape of the API will be able to identify an openfga server.|
+|Impact (High/ Med/ Low)|Low|
+|Likelihood (High/ Med/ Low)|Low|
+
+|Summary|Usage of PII as part of models.|
+|--|--|
+|Discovered in self-assessment?|Yes|
+|Weakness|PII can be exfiltrated if present as a part of the models.|
+|MITRE classification|TA0042: Resource Development -> T1586: Compromise Accounts|
+|Actors|openfga.server|
+|Suggested Mitigation|Specifically called out in documentation that PII is not to be used as a part of models but not enforced.<br>Also, many sites will use email ID as primary login, documentation may need to address how to handle this situation.<br>A change to detect PII such as email ID's as part of models and relationships/ tuples and warn the user may tackle this finding at a code level.|
+|Impact (High/ Med/ Low)|Med|
+|Likelihood (High/ Med/ Low)|Low|
+
+|Summary|Turning off audit logs will let an attacker mask operations on the server.|
+|--|--|
+|Discovered in self-assessment?|No|
+|Weakness|Allowing audit logs to be turned off completely via configuration would allow attackers to mask their trail.|
+|MITRE classification|TA0005: Defense Evasion -> T1562: Impair Defenses|
+|Actors|openfga.server|
+|Suggested Mitigation|A possible mitigation is to log a warning message when logs are completely turned off?|
+|Impact (High/ Med/ Low)|Med|
+|Likelihood (High/ Med/ Low)|Low|
+
+|Summary|DDOS attack on openfga server would impact overall application landscape.|
+|--|--|
+|Discovered in self-assessment?|Yes|
+|Weakness|A DDOS attack on the server endpoint is not handled by the server and will result in widespread impact on all dependent applications.|
+|MITRE classification|TA0005: Defense Evasion -> T1562: Impair Defenses|
+|Actors|openfga.server|
+|Suggested Mitigation|Documentation can be created for production configuration using an API Gateway/ other DDOS prevention mechanism.|
```

Comment:
Given that the server should not be exposed to the public internet, my understanding is that the DDOS can only happen if the application itself is DDOS'd.

In any case, we can add documentation on how to protect the OpenFGA server https://github.com/openfga/openfga.dev/issues/775